lib/runners/processor/actionlint.rb

```
@@ -0,0 +1,87 @@
+module Runners
+  class Processor::Actionlint < Processor
+    SCHEMA = _ = StrongJSON.new do
+      extend Schema::ConfigTypes
+
+      # @type self: SchemaClass
+      let :config, base(
+        target: target,
+
+        # Native options
+        ignore: one_or_more_strings?,
+      )
+    end
+
+    ALLOWED_ERRORS = [
+      "no project was found in any parent directories of",
+      "no YAML file was found in",
+    ].freeze
+
+    register_config_schema SCHEMA.config
+
+    def self.config_example
+      <<~'YAML'
+        root_dir: project/
+        target:
+          - .github/workflows/one_workflow.yml
+          - .github/workflows/another_workflow.yml
+        ignore:
+          - "job .* needs job"
+          - "unknown Webhook event"
+      YAML
+    end
+
+    # This analyser uses git metadata (.git/).
+    def use_git_metadata_dir?
+      true
+    end
+
+    def analyzer_options
+      [].tap do |opts|
+        opts << "-format"
+        opts << "{{json .}}"
+
+        Array(config_linter[:ignore]).each { |v| opts << "-ignore" << v }
+
+        opts << "--"
+        Array(config_linter[:target]).each { |v| opts << v }
+      end
+    end
+
+    def analyze(changes)
+      stdout, stderr, status = capture3(
+        analyzer_bin,
+        *analyzer_options,
+      )
+
+
+      if ALLOWED_ERRORS.any? { |pattern| stderr.include?(pattern) }
+        add_warning(stderr)
+        return Results::Success.new(guid: guid, analyzer: analyzer)
+      elsif not [0, 1].include?(status.exitstatus)
+        add_warning(stderr)
+        return Results::Failure.new(guid: guid, analyzer: analyzer)
```

In this case, the error message should be stored in the `Failure` object rather than warnings.
But I'm not sure if the error message is human readable. Can you reproduce any error just in case?

---

The errors that I was able to reproduce were:
```shell
no project was found in any parent directories of ".". check workflows directory is put correctly in your Git repository

no YAML file was found in "/home/rado/temp/new/.github/workflows"

could not read "nonexistent_file": open nonexistent_file: no such file or directory

$ actionlint -nonexistentflag
flag provided but not defined: -nonexistentflag
Usage: actionlint [FLAGS] [FILES...] [-]

  actionlint is a linter for GitHub Actions workflow files.
...[the whole help string]
  -version
    	Show version and how this binary was installed
```

---

The first two actually are categorized as Success. As for others, I cannot guarantee that `stdout` will be human-readable.

---

[actionlint documentation](https://github.com/rhysd/actionlint/blob/main/docs/usage.md) does not provide list of all possible stderr outputs, so it's hard to determine if they are usable as our failure message.

The exit code also does not give enough information for that:

Status | Description
-- | --
0 | The command ran successfully and no problem was found
1 | The command ran successfully and some problem was found
2 | The command failed due to invalid command line option
3 | The command failed due to some fatal error

So I would be in favor of having a fixed array of error messages that could be used in the stdout as they are. If `stderr` does not match any of those, the user would have to look into the logs.

Do you have any specific error messages that you would like to use for `Results::Failure.new(message: ...)`?

---

Users cannot pass program arguments, so the forth is negligible. As for the third, it would be useful for users to identify the cause of the error.
I think we can provide `stderr` as an error message. Also, we have `trace_writer.error` method to put messages in the trace log. The codes will go to;

```ruby
      elsif not [0, 1].include?(status.exitstatus)
        trace_writer.error stderr <- this logs detailed message.
        return Results::Failure.new(guid: guid, analyzer: analyzer, message: headline(stderr)) <- reports the first line.
      end

def headline(strerr)
      if !stderr.empty?
        stderr.lines(chomp: true).first
      else
        nil
      end
end
```

---

Upon @ybiquitous 's suggestion, I'm not raising an error instead of returning `Results::Failure`, so I'm not sure if now producing the error headline makes sense. stderr is in the logs anyway and `raise` message does not appear in the webUI anyway. 

By the way, is the `Results::Failure`'s `message:` argument visible to the user somewhere?